### ISSUE TITLE
fun with return type Nothing is not shown

### ISSUE THREAD

**Issue by robstoll** [Kotlin/dokka#262]
I have overloads for one functions (placed in two different files, maybe that matters).
The first looks as follows (shortened) 
```
/** my fun
 */
fun <T: Any> A<T>.foo(expected: T): A<T> { ... }
```
and another as follows:
```
@Deprecated("bla bla, ReplaceWith("bar(expected)"))
/** my fun
 */
fun <:BigDecimal> A<T>.foo(): Nothing = throw UnsupportedOperationException("...")
```
Unfortunately the second one is not shown in the output. It does not seem to be due to `@Deprecated` so I guess it is due to `Nothing`

**Comment by semoro** [comment]
Please, make sure that skipDeprecated = false in your setup
as I can't reproduce that
I used this sample:
```
package foo

import java.math.BigDecimal

class A<T>

/** my fun
 */
fun <T: Any> A<T>.foo(expected: T): A<T> { ... }

/** my fun
 */
@Deprecated("bla bla", ReplaceWith("bar(expected)"))
fun <T: BigDecimal> A<T>.foo(): Nothing = throw UnsupportedOperationException("...")
```

**Comment by semoro** [comment]
Is it still reproduces?

**Comment by robstoll** [comment]
Yes, seems still to be the case (with 0.9.16). Consider:
https://robstoll.github.io/atrium/0.6.0/doc/ch.tutteli.atrium.api.cc.en_-u-k/to-be.html

There is just one overload, the one which is missing is the following:
https://github.com/robstoll/atrium/blob/v0.6.0/apis/atrium-api-cc-en_UK/src/main/kotlin/ch/tutteli/atrium/api/cc/en_UK/bigDecimalAssertions.kt#L16

**Comment by Kordyjan** [comment]
As I understand, this issue should be fixed in 1.4.0. If it is still there feel free to reopen it.
